warp10/src/main/java/io/warp10/WarpURLDecoder.java

```diff
@@ -0,0 +1,41 @@
+//
+//   Copyright 2019  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.Charset;
+
+public class WarpURLDecoder {
+  public static String decode(String input, Charset charset) throws UnsupportedEncodingException {
+    
+    // We only decode the % encoding.
+    
+    if (!input.contains("%")) {
+      return input;
+    }
+    
+    // The input contains a '%', if it contains a '+' we need to replace the '+' with '%2B'
+    // so decoding is correct.
+    
+    if (input.contains("+")) {
+      input = input.replaceAll("+", "%2B");
+    }
```

Comment:
Use input.replaceAll("\\+", "%2B") or input.replace() which is optimized for literal targets, otherwise it can't work.

Test with this WarpScript:
```
  '.*[!%25&\(\)\*+/<=>\[\]^\{\|\}~].*'  MATCHER 'ForbiddenCharacters' STORE
```
You will have exception `Dangling meta character '+' near index 0 + ^`